logstash-core/spec/conditionals_spec.rb

```diff
@@ -51,27 +51,40 @@ def multi_receive(events)
     end
   end
 
-  before do
-    LogStash::PLUGIN_REGISTRY.add(:output, "dummynull", DummyNullOutput)
-  end
-
   describe "simple" do
-    config <<-CONFIG
-      input {
-        generator {
-          message => '{"foo":{"bar"},"baz": "quux"}'
-          count => 1
-        }
+    let(:config) do <<-CONFIG
+    input {
+      generator {
+        message => '{"foo":{"bar"},"baz": "quux"}'
+        count => 1
       }
-      output {
-        if [foo] == "bar" {
-          dummynull { }
-        }
+    }
+    output {
+      if [foo] == "bar" {
+        dummynull { }
       }
+    }
     CONFIG
+    end
+
+    let(:pipeline) do
+      settings = ::LogStash::SETTINGS.clone
+      config_part = org.logstash.common.SourceWithMetadata.new("config_string", "config_string", config)
+      pipeline_config = LogStash::Config::PipelineConfig.new(LogStash::Config::Source::Local, :main, config_part, settings)
+      LogStash::JavaPipeline.new(pipeline_config)
+    end
+
+    before do
+      LogStash::PLUGIN_REGISTRY.add(:output, "dummynull", DummyNullOutput)
+    end
+
+    after do
+      pipeline.close
+    end
 
-    agent do
+    it "example run" do
```

Comment:
It's not clean to me what was the meaning of the previous
```ruby
agent do
end
```

probably we have to test that an event is correctly processed by `output` section (like for the filters one in "conditionals in filter"), but since this wasn't covered by the test before, it could go in a follow up PR

Reply:
The original one tested the `pipeline.run`, and make sure no error and exception.
I can rename it to be more clear.

Reply:
"example run" is ok, it wasn't clear to me why the difference between tests